lib/rules/array-callback-return.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @fileoverview Rule to enforce return statements in callbacks of array's methods
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+var TARGET_NODE_TYPE = /^(?:Arrow)?FunctionExpression$/;
+var TARGET_METHODS = /^(?:every|fi(?:lter|nd(?:Index)?)|map|reduce(?:Right)?|so(?:me|rt))$/;
```

Comment:
Why didn't you compare against the most obvious approach of `x === "a" || x === "b" || ...`?